src/core/core.scale.js

```diff
@@ -116,6 +116,48 @@ function parseLineHeight(options) {
 		helpers.valueOrDefault(options.fontSize, defaults.global.defaultFontSize));
 }
 
+function fillTextVertical(context, text, x, y) {
+	var arrText = text.split('');
+	var arrWidth = arrText.map(function(letter) {
+		return context.measureText(letter).width;
+	});
+
+	context.translate(0, 0);
+	var align = context.textAlign;
+	var baseline = context.textBaseline;
+
+	if (align === 'left') {
+		x = x + Math.max.apply(null, arrWidth) / 2;
+	} else if (align === 'right') {
+		x = x - Math.max.apply(null, arrWidth) / 2;
+	}
+	if (baseline === 'bottom' || baseline === 'alphabetic' || baseline === 'ideographic') {
+		y = y - arrWidth[0] / 2;
+	} else if (baseline === 'top' || baseline === 'hanging') {
+		y = y + arrWidth[0] / 2;
+	}
+
+	context.textAlign = 'center';
+	context.textBaseline = 'middle';
+	arrText.forEach(function(letter, index) {
+		context.setTransform(1, 0, 0, 1, 0, 0);
+		var letterWidth = arrWidth[index];
+		var code = letter.charCodeAt(0);
+		if (code <= 256) {
```

Comment:
why are you treating ascii and non-ascii characters differently? this should have a comment to explain

Reply:
Agreed. It sounds like this should be a new feature in it's own right as I can see english text wanting this behaviour.